common-libs/common/src/shared/DidResolver.ts

```diff
@@ -17,42 +24,70 @@ const getService = (options: ConstructorOptions) => {
   if (!services.has(cacheKey)) {
     services.set(cacheKey, {
       service: new RegistryApiService(options),
-      cache: new Map(),
+      cache: options.useCache
+        ? new LRUCache({
+            max: options.cacheMaxSize ?? DEFAULT_CACHE_MAX_SIZE,
+            ttl: (options.cacheTtlInMin ?? DEFAULT_CACHE_TTL_IN_MIN) * 60 * 1000,
+            allowStale: true,
+          })
+        : undefined,
     })
   }
 
   return services.get(cacheKey)
 }
 
 const resolveDid = ({ service, cache }: ServiceWithCache, did: string) => {
-  if (!cache.has(did)) {
-    const promise = new Promise((resolve, reject) => {
-      service
-        .resolveDid({ did })
-        .then(({ body }) => {
-          const { didDocument } = body
-          resolve(didDocument)
-        })
-        .catch((reason) => {
-          // Don't cache fails
-          cache.delete(did)
-          reject(reason)
-        })
-    })
-    cache.set(did, promise as Promise<DidDocument>)
+  if (cache.has(did)) {
+    return cache.get(did)
   }
 
+  const promise = new Promise((resolve, reject) => {
+    service
+      .resolveDid({ did })
+      .then(({ body }) => {
+        const { didDocument } = body
+        resolve(didDocument)
+      })
+      .catch((reason) => {
+        // Don't cache fails
+        cache.delete(did)
+        reject(reason)
+      })
+  })
+  cache.set(did, promise as Promise<DidDocument>)
   return cache.get(did)
 }
 
-export class DidResolver {
+const resolveDidWithoutCache = ({ service }: ServiceWithCache, did: string) => {
```

Comment:
Do not wrap in additional Promise this simple call
```

const { didDocument } = await service.resolveDid({ did })

return didDocument

```

Reply:
But return value for resolveDid is Promise<DidDocument>, line 96
`resolveDid: (did: string) => Promise<DidDocument>`